submariner/multiple-cable-drivers.md

```diff
@@ -0,0 +1,415 @@
+# Support fine-grained cluster connectivity and multiple cable drivers
+
+## Summary
+
+Submariner currently assumes full mesh connectivity by default and provisions VPN connections between all participating
+clusters. In addition, Submariner currently offers one of three cable drivers: VXLAN, IPSec or wireguard with limited
+configuration options. Only one of these cable drivers is enabled across all the clusters in a cluster set, that is all
+clusters that join a broker must use the same cable driver. The goal of this proposal is to outline an extended model
+that will support:
+
+- The selection of which clusters to connect and in what fashion (full mesh or connection to subset of clusters in
+  a ClusterSet).
+- The selection of one or more cable drivers in order to support different cable types between connected clusters.
+  For example, IPSec between public and private clouds but raw VXLAN between the clusters in the same cloud.
+
+There are use cases that involve supporting a mix of cable drivers or cable driver configurations within the same
+cluster set. For example, a cluster set could contain three clusters, two that are on-premise and the other
+in the cloud. Connections to the cloud cluster will need to use encryption while the on-premise clusters can use
+unencrypted connections, as they’re connected by a private link, to avoid the performance overhead of encryption.
+This could be achieved using either:
+
+- Different cable driver implementations or
+- The same cable driver implementation with differing configurations.
+
+For other use cases, it makes sense to offer finer-grained connectivity options. One such use case is a client-server
+scenario where multiple client clusters need to access a server cluster but the user does not want the client clusters
```

Comment:
Doesn't sound like Submariner is a good fit for this use case.

Reply:
First, secure tunnels are point to point by design, there should be no middlemen eavesdropping on the contents, so I'm not sure how you expect to provide a secure tunnel with a hub cluster.

Second, I think where clusters on "the edge" don't need to talk to each other, call them clients or whatever, then this isn't what Submariner is trying to solve.